argo/providers/vai/eventing_server.go

```diff
@@ -107,9 +108,13 @@ func modelServingArtifactsForJob(job *aiplatformpb.PipelineJob) []ServingModelAr
 		for name, output := range task.GetOutputs() {
 			for _, artifact := range output.GetArtifacts() {
 				if artifact.SchemaTitle == PushedModelArtifactType {
-					pushedProperty, ok := artifact.Metadata.AsMap()[ModelPushedMetadataProperty]
-					if ok && pushedProperty.(float64) == ModelPushedMetadataValue {
-						servingModelArtifacts = append(servingModelArtifacts, ServingModelArtifact{Name: name, Location: artifact.GetUri()})
+					properties := artifact.Metadata.AsMap()
+					pushedProperty, hasPushed := properties[ModelPushedMetadataProperty]
+					if hasPushed && pushedProperty.(float64) == ModelPushedMetadataValue {
+						pushedDestinationProperty, hasPushedDestination := properties[ModelPushedDestinationProperty]
+						if hasPushedDestination {
+							servingModelArtifacts = append(servingModelArtifacts, ServingModelArtifact{Name: name, Location: pushedDestinationProperty.(string)})
```

Comment:
we should check that the property is actually a string or else we risk a runtime panic. This hasn't been done for now because we are not doing it in line 113 either. I'd be happy to add it (for both cases)

Reply:
I was checking how to do that to leave a comment. I think we should.

Reply:
this is now done